lib/AIECreateLocks.cpp

```diff
@@ -51,9 +55,12 @@ struct Token2LockLowering : public OpConversionPattern<UseTokenOp> {
       srcCol = mem.colIndex();
       srcRow = mem.rowIndex();
       IsParentMemOp = true;
+    } else if (auto shimDma = dyn_cast<ShimDMAOp>(parentOp)) {
+      srcCol = shimDma.colIndex();
+      srcRow = shimDma.rowIndex();
     } else {
-      llvm_unreachable(
-          "A parent operation of UseTokenOp must be either CoreOp or MemOp");
+      llvm_unreachable("A parent operation of UseTokenOp must be either CoreOp "
+                       "or MemOp or ShimDMAOp");
```

Comment:
probably this should be moved to a verifier?  Actually I was thinking we should extract all of the 'token' stuff into a separate dialect entirely.

Reply:
Added a verifier for UseTokenOp.